src/Processors/Transforms/RemoteDependencyTransform.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <Processors/IProcessor.h>
+#include <Processors/ResizeProcessor.h>
+
+namespace DB
+{
+
+class RemoteQueryExecutor;
+using RemoteQueryExecutorPtr = std::shared_ptr<RemoteQueryExecutor>;
+
+class ReadFromMergeTreeDependencyTransform : public IProcessor
+{
+public:
+    explicit ReadFromMergeTreeDependencyTransform(const Block & header);
+
+    String getName() const override { return "ReadFromMergeTreeDependency"; }
+    Status prepare() override;
+
+    InputPort & getInputPort() { assert(data_port); return *data_port; }
+    InputPort & getDependencyPort() { assert(dependency_port); return *dependency_port; }
+    OutputPort & getOutputPort() { return outputs.front(); }
+
+    void connectToScheduler(ResizeProcessor & scheduler);
```

Comment:
Let's maybe replace resize_processor to `OutputPort &` ? And remove `ResizeProcessor::getFreeOutputPortIfAny`